File: java-start/app/src/main/java/com/vonage/tutorial/voice/BaseApplication.java
package com.vonage.tutorial.voice;

import android.app.Application;
import com.nexmo.client.BuildConfig;
import com.vonage.tutorial.voice.util.Tutorial;
import timber.log.Timber;

public final class BaseApplication extends Application {

    public void onCreate() {
        super.onCreate();

        if (BuildConfig.DEBUG) {
            Timber.plant(new Timber.DebugTree());
        }

        this.initializeNexmoClient();
    }

    private final void initializeNexmoClient() {
        // Init the NexmoClient. You can retrieve NexmoClient instance latter by using NexmoClient.get()
        Tutorial.todo("Init NexmoClient here");
    }
}

File: kotlin-start/app/src/main/java/com/vonage/tutorial/voice/BaseApplication.kt
package com.vonage.tutorial.voice

import android.app.Application
import com.nexmo.client.BuildConfig
import timber.log.Timber

class BaseApplication : Application() {

    override fun onCreate() {
        super.onCreate()

        if (BuildConfig.DEBUG) {
            Timber.plant(Timber.DebugTree())
        }

        initializeNexmoClient()
    }

    private fun initializeNexmoClient() {
        // Init the NexmoClient. You can retrieve NexmoClient instance latter by using NexmoClient.get()
        TODO("Init NexmoClient here")
    }
}